Runtime/Extensions/CollectionsExtensions.cs

```diff
@@ -199,5 +189,24 @@ public static void EnsureListItems<T>(this IList<T> list, IList<T> items)
             }
             return false;
         }
+
+        /// <summary>
+        /// Safely removes an item from a <see cref="Dictionary{TKey, TValue}"/> if it is found.
+        /// </summary>
+        /// <typeparam name="TKey">Data type used in the Dictionary Key.</typeparam>
+        /// <typeparam name="TValue">Data type used in the Dictionary Value.</typeparam>
+        /// <param name="dictionary">The instance of the <see cref="Dictionary{TKey, TValue}"/> to validate.</param>
+        /// <param name="key">The Key of a <see cref="KeyValuePair{TKey, TValue}"/> to validate against the dictionary with.</param>
+        /// <returns>True if an item was removed from the collection</returns>
+        public static bool SafeRemoveDictionaryItem<TKey, TValue>(this Dictionary<TKey, TValue> dictionary, TKey key)
+        {
+            if (dictionary.TryGetValue(key, out _))
```

Comment:
Should this not use contains instead?

Reply:
TryGet achieves the same thing and is safer